pandas/core/series.py

```diff
@@ -623,6 +623,29 @@ def view(self, dtype=None):
         return self._constructor(self._values.view(dtype),
                                  index=self.index).__finalize__(self)
 
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        inputs = tuple(
+            x._values if isinstance(x, type(self)) else x
+            for x in inputs
+        )
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not None:
+            return result
+
+        if hasattr(self._values, '__array_ufunc__'):
+            result = self._values.__array_ufunc__(
+                ufunc, method, *inputs, **kwargs)
+        else:
+            result = np.array(self._values).__array_ufunc__(
+                ufunc, method, *inputs, **kwargs)
+        if result is NotImplemented:
+            raise TypeError("The '{0}' operation is not supported for "
+                            "dtype {1}.".format(ufunc.__name__, self.dtype))
```

Comment:
Instead of all this (checking for `__array_ufunc__` and then the branch/error message), you could just reinvoke the public ufunc again (as I suggested for IntegerArray). That would also make me more confidence that the proper delegation is happening.

Reply:
Thanks, that sounds like a better idea :)